tests/drivers/abp2/main.c

```diff
@@ -0,0 +1,126 @@
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the ABP2 Honeywell ABP2 series
+ *              pressure and temperature sensor driver.
+ *
+ * @author      David Picard <david.picard@clermont.in2p3.fr>
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "ztimer.h"
+#include "timex.h"
+#include "phydat.h"
+#include "abp2.h"
+#include "abp2_params.h"
+
+#define STATUS_SLEEP_MS         (1)
+#define MEASUREMENT_SLEEP_MS    (1000)
+#define MAX_LOOPS_STATUS        (10)
+#define MAX_LOOPS_MEAS          (5)
+
+static abp2_t dev;
+
+int main(void)
+{
+    int32_t press = 0;
+    int32_t temp = 0;
+    abp2_params_t prms;
+    phydat_t phyPress;
+    phydat_t phyTemp;
+    void *ptr;
+    int blockingMode = 1;
+    int cntMeas = 0;
+
+    puts("ABP2 Honeywell series pressure and temperature sensor\n");
+
+    ptr = memcpy(&prms, &abp2_params, sizeof(abp2_params));
+    if(!ptr)
+    {
+       puts("main() >> ERROR: memcpy() failed");
+       return 1;
+    }
+
+    printf("Initializing SPI bus %d\n", prms.spi);
+    spi_init(prms.spi);
+
+    printf("Initializing ABP2 at SPI_DEV(%i)... ", prms.spi);
+
+    if (abp2_init(&dev, &prms) == 0)
+    {
+        puts("[OK]");
+    }
+    else {
+        puts("[Failed]");
+        return -1;
+    }
+
+    puts("=========================");
+    puts("        Measuring");
+    puts("=========================");
+
+    printf("Pressure range = %d .. %d\n", (int)prms.rangeMin, (int)prms.rangeMax);
+
+    while (1) {
+        int res;
+
+        ztimer_sleep(ZTIMER_MSEC, MEASUREMENT_SLEEP_MS);
+
+        if (blockingMode) {
+            res = abp2_read(&dev, &press, &temp);
+            if (res)
+            {
+                printf("abp2_read() >> ERROR errno = %d", res);
+                continue;
+            }
+        }
+        else {
+            res = abp2_read_nb(&dev, &press, &temp);
+            if (res)
+            {
+                printf("abp2_read_nb() >> ERROR errno = %d", res);
+                continue;
+            }
+        }
+        /* printf() doesn't support %f on all boards. This is where phydat comes in handy.
+         * The default range in RIOT/drivers/abp2/include/abp2_params.h is 0..160000ubar = 0..160mbar.
+         */
+
+        phyPress.val[0] = press;                /* press is already in ubar */
+        phyPress.scale = -6;                    /* 1 ubar = 1e-06 bar */
+        phyPress.unit = UNIT_BAR;               /* set the unit */
+        phydat_dump(&phyPress, 1);              /* print the value in a pretty format */
+
+        phyTemp.val[0] = temp;                  /* temp is already in mdeg C */
+        phyTemp.scale = -3;                     /* 1 mdegC = 1e-03 degC */
+        phyTemp.unit = UNIT_TEMP_C;             /* set the unit */
+        phydat_dump(&phyTemp, 1);               /* print the value in a pretty format */
```

Comment:
shouldn't the test make use of what is implemented in abp2_saul.c

Reply:
Fair enough. I'm working on it.

Reply:
OK, done. I updated the serial output at the top of this PR. I only found 1 test, demonstrating the SAUL interface...

```bash
$ cd tests/drivers/
$ grep -F "saul_reg" $(find . -name main.c)
```